### PR TITLE
chore: monorepo structure and new ci for releases

### PR DESCRIPTION
**Pull request by hugomrdias**
check commit 0004747 for ci and code changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-    "build:client:docs": "yarn --cwd client typedoc",
-    "build:website": "yarn --cwd website build",
-    "format": "prettier --write '**/*.{js,jsx,ts,tsx,md,html,css}' --ignore-path .gitignore"
+    "test": "run-p test:*",
+    "test:client": "yarn --cwd packages/client test",
+    "test:api": "yarn --cwd packages/api test",
+    "test:website": "yarn --cwd packages/website test",
+    "build:client:docs": "yarn --cwd packages/client typedoc",
+    "build:website": "yarn --cwd packages/website build",
+    "format": "prettier --write '**/*.{js,jsx,ts,tsx,md,html,css}' --ignore-path .gitignore",
```
Comment (alanshaw) [inline package.json:13]: I pushed a change to main to add a `.prettierignore` as it was trying to reformat the docs directory which is not gitignored.

```suggestion
    "format": "prettier --write '**/*.{js,jsx,ts,tsx,md,html,css}'",
```

File: packages/tools/cli.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  .option('--ttl', 'Record TTL', 1)
+  .option('--proxied', 'Record should be proxied ?', true)
+  .action(async (/** @type {import('./types').UpsertRecordOptions} */ opts) => {
+    await upsertRecord(opts)
+  })
+
+cli
+  .command(
+    'deploy-website',
+    'Update dns record with latest Cloudflare Pages production deployment.'
```
Comment (alanshaw) [inline packages/tools/cli.js:61]:
```suggestion
    'Update DNS record with latest Cloudflare Pages production deployment.'
```

File: packages/tools/cli.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        owner: 'ipfs-shipyard',
+        repo: 'nft.storage',
+        ref: 'main',
+        check_name: 'Cloudflare Pages',
+        status: 'completed',
+      })
+      const conclusion = data.check_runs[0].conclusion
+      if (conclusion === 'success') {
+        const url = [...getUrls(data.check_runs[0].output.summary)][0]
+        if (url.endsWith('nft-storage.pages.dev')) {
```
Comment (alanshaw) [inline packages/tools/cli.js:82]: Could be `undefined` if no URLs:

```suggestion
        if (url && url.endsWith('nft-storage.pages.dev')) {
```

File: packages/tools/cli.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            zone: opts.zone,
+            name: opts.name,
+            type: opts.type,
+            ttl: opts.ttl,
+            proxied: opts.proxied,
+            content: url.replace('https://', ''),
+          })
+        } else {
+          throw new Error(
+            `URL is invalid "${url}" should end with  nft-storage.pages.dev`
```
Comment (alanshaw) [inline packages/tools/cli.js:95]:
```suggestion
            `URL is invalid "${url}" should end with nft-storage.pages.dev`
```

**Review by alanshaw — CHANGES_REQUESTED** [review]
🚀 looks awesome, I've highlighted some minor tweaks and fixes. Please would you update the README to document the release flow? Including the fact that the release PR will be updated with additional commits to main (so no need to close if there's more stuff going tinto the release) and that we should be using conventional commits and flagging breaking changes using "!".

i.e. PR -> Merge -> github action release PR -> Merge -> github action deploy/publish

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            nodePaths = Array.from(nodePaths);
+            if(nodePaths.length === 0){
+              console.log(`::warning::No changes found, release will be skipped.`)
+            }
+            console.log(`::set-output name=nodePaths::${JSON.stringify(nodePaths)}`);
+
+  release-pr:
+    runs-on: ubuntu-latest
+    needs: changeFinder
+    if: ${{ fromJson(needs.changeFinder.outputs.nodePaths)[0] != '' }}
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply .github/workflows/release.yml:46]: its github syntax.. seems to coerce to what we need here, which is "do not run this when we change something outside the packages folder"

File: README.md [hunk 6]
```diff
@@ -236,6 +239,37 @@ yarn install
 yarn dev
 ```
 
+## Release
+
+Release Please automates CHANGELOG generation, the creation of GitHub releases,
```
Comment (alanshaw) [inline README.md:244]:
```suggestion
[Release Please](https://github.com/googleapis/release-please) automates CHANGELOG generation, the creation of GitHub releases,
```

**Review by alanshaw — APPROVED** [review]

**Pull request merged**